StationeersLaunchPad/ModLoader.cs

```diff
@@ -146,15 +147,30 @@ public static List<Type> FindBepInExEntrypoints(List<Assembly> assemblies)
       var result = new List<(Type, MethodInfo)>();
       foreach (var assembly in assemblies)
       {
+        var failed = true;
+
         var name = assembly.GetName().Name;
-        var defType = assembly.GetType(name);
-        if (defType == null || !typeof(MonoBehaviour).IsAssignableFrom(defType))
-          continue;
+        var type = assembly.GetType(name);
+        if (type == null || type.IsAbstract || !typeof(MonoBehaviour).IsAssignableFrom(type))
+          failed = true;
 
-        var defMethod = defType.GetMethod(DEFAULT_METHOD, BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, new Type[] { typeof(List<GameObject>) }, null);
-        if (defMethod == null || result.Contains((defType, defMethod)))
-          continue;
-        result.Add((defType, defMethod));
+        var method = type?.GetMethod(DEFAULT_METHOD, BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, new Type[] { typeof(List<GameObject>) }, null);
+        if (method == null || result.Contains((type, method)))
+          failed = true;
+
+        if (failed)
```

Comment:
if we succeed in finding the class matching the assembly name, we won't end up adding it to the results right?